crates/extensions/tedge-p11-server/src/main.rs

```diff
@@ -83,14 +85,34 @@ fn main() -> anyhow::Result<()> {
         }
         if let Some((name, fd)) = systemd_listeners.next() {
             debug!(?name, "Using socket passed by systemd");
-            UnixListener::from(fd)
+            let listener = UnixListener::from(fd);
+            (listener, None)
         } else {
             debug!("No sockets from systemd, creating a standalone socket");
-            UnixListener::bind(socket_path).context("Failed to bind to socket")?
+            let listener = UnixListener::bind(&socket_path)
+                .with_context(|| format!("Failed to bind to socket at '{socket_path}'"))?;
+            (
+                listener,
+                Some(OwnedSocketFileDropGuard(socket_path.clone())),
+            )
         }
     };
     info!(listener = ?listener.local_addr().as_ref().ok().and_then(|s| s.as_pathname()), "Server listening");
-    TedgeP11Server::from_config(cryptoki_config)?.serve(listener)?;
+    let listener = tokio::net::UnixListener::from_std(listener)?;
+    let server = TedgeP11Server::from_config(cryptoki_config)?;
+    tokio::spawn(async move { server.serve(listener).await });
+
+    // by capturing SIGINT, we allow owned socket drop guard to run before exit
+    tokio::signal::ctrl_c().await?;
```

Comment:
`SIGTERM` has to be handled too as this is the signal used in practice to kill a daemon.

Reply:
Fixed by waiting for `SIGINT` or `SIGTERM` concurrently.